dissect/target/plugins/os/windows/generic.py

```diff
@@ -123,11 +123,7 @@ def check_compatible(self):
     @export(property=True)
     def ntversion(self):
```

Comment:
Is it worth promoting `nt_version` to a full `@export`, and adding a deprecation warning or maybe just outright remove this one?

Reply:
Now that you mention it, it should probably be the other way around. The idea was to have the interface be the same for all OSplugin implementations so you know you can always call `target.<plugin>` irrespective of OS. This would mean these functions should not be `@internal` and possibly should get a `_` prepended to their name.

Reply:
I re-implemented this a bit, which meant I needed some facilities which were not yet available (access to the OSPlugin instance for the target). I implemented these in #129 .

Reply:
Becasue the `_os.py` plugin should just implement the interface defined by the `OSPlugin` base class. The idea behind this is that the rest of the code gets the guarantee that functions here are always available and implemented once a target is fully loaded and has an OS assigned. This means the functionality should be applicable to all OSes supported this way, all other stuff should go into generic.py.

Note:
In practice the implementation requirement can be somewhat circumvented as certain functions allow `None` to be returned as certain functionality may be missing in broken targets. This means these functions could be implemented as stubs just retruning `None`.

Note 2:
Also there are 2 functions in the windows `_os.py` that violate this: `codepage()` which should be moved to `generic.py` and `architecture()`, which should become a proper `OSPlugin` function.